components/TemplateGrid/TemplateDesigner/TemplateGridDraggableLayer.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+    <div class="draggable-layer">
+        <TemplateGridGhostItem
+            v-for="(position, index) in gridLayerPositions"
+            :key="index"
+            :style="{ gridArea: `${position.row} / ${position.column} / auto / auto` }"
+            :position="position"
+            :highlighting-positions="highlightedPositions"
+            :min-highlighted-column="minHighlightedColumn"
+            :max-highlighted-column="maxHighlightedColumn"
+            :min-highlighted-row="minHighlightedRow"
+            :max-highlighted-row="maxHighlightedRow"
+            @drop="onDrop" />
+        <LayoutElement
+            v-for="(element, index) in layoutElements"
+            :key="`${element.row}/${element.column}`"
+            :style="getGhostItemPosition(element)"
+            :index="index"
+            :element="element"
+            @highlightedPositionChange="onHighlightedPositionsChange" />
+    </div>
+</template>
+
+<script>
+import {
+    maxObjectValueInArrayByObjectKey,
+    minObjectValueInArrayByObjectKey,
+} from '~/model/arrayWrapper';
+
+export default {
+    name: 'TemplateGridDraggableLayer',
+    components: {
+        TemplateGridGhostItem: () => import('~/components/TemplateGrid/TemplateDesigner/TemplateGridGhostItem'),
+        LayoutElement: () => import('~/components/Template/TemplateDesigner/LayoutElement'),
+    },
+    props: {
+        columnsNumber: {
+            type: Number,
+            required: true,
+        },
+        rowsNumber: {
+            type: Number,
+            required: true,
+        },
+        layoutElements: {
+            type: Array,
+            required: false,
+            default: () => [],
+        },
+    },
+    data() {
+        return {
+            highlightedPositions: [],
+            minHighlightedRow: 0,
+            maxHighlightedRow: 0,
+            minHighlightedColumn: 0,
+            maxHighlightedColumn: 0,
+        };
+    },
+    computed: {
+        gridLayerPositions() {
+            const length = this.rowsNumber * this.columnsNumber;
+            const positions = [];
+
+            for (let i = 1; i <= length; i += 1) {
+                const row = Math.ceil(i / this.columnsNumber);
+                const tmpColumn = i % this.columnsNumber;
+                const column = tmpColumn !== 0 ? tmpColumn : this.columnsNumber;
+
+                positions.push({
+                    row,
+                    column,
+                });
+            }
+
+            return positions;
+        },
+    },
+    methods: {
+        onDrop(position) {
+            this.$emit('addListElementToLayout', position);
+        },
+        onHighlightedPositionsChange(positions) {
+            this.highlightedPositions = positions;
+
+            this.maxHighlightedRow = maxObjectValueInArrayByObjectKey(positions, 'row');
```

Comment:
maxObjectValueInArrayByObjectKey, minObjectValueInArrayByObjectKey - names should be more descriptive in case of what they returns

Reply:
They return min and max value - the name is pointing on it - isint it ?

Reply:
Yes, but they looks more like variables than a functions, I think they should start with "find", "get" etc...